### PR TITLE
Feat registry consul

### LINKED ISSUE #379: higress对接consul注册中心
## Why do you need it?
 Is your feature request related to a problem? Please describe in details

## How could it be?
A clear and concise description of what you want to happen. You can explain more about input of the feature, and output of it.

### PR DESCRIPTION
**Pull request by 2456868764**
### Ⅰ. Describe what this PR did

add consul registry  and support Nacos & Consul services with authentication enabled.

### Ⅱ. Does this pull request fix one issue?

fixes #379 , #438

### CODE REVIEW

**Review by johnlanni — CHANGES_REQUESTED** [review]
Please add an e2e test for consul registry

**Comment by 2456868764** [comment]
i‘ve added an e2e test for consul registry, please review it again

File: test/e2e/conformance/tests/httproute-consul-demo.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import (
+	"testing"
+
+	"github.com/alibaba/higress/test/e2e/conformance/utils/http"
+	"github.com/alibaba/higress/test/e2e/conformance/utils/suite"
+)
+
+func init() {
+	HigressConformanceTests = append(HigressConformanceTests, HTTPRouteHttp2Rpc)
```
Comment (Xunzhuo) [inline test/e2e/conformance/tests/httproute-consul-demo.go:25]: I think here is a mistake.

File: test/e2e/conformance/tests/httproute-consul-demo.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+var HTTPRouteConsulDemo = suite.ConformanceTest{
```
Comment (Xunzhuo) [inline test/e2e/conformance/tests/httproute-consul-demo.go:28]: HTTPRouteConsulDemo is not a good name to demonstrate what this case covers.

File: test/e2e/conformance/tests/httproute-consul-demo.go (hunk 1 above)
Comment (2456868764) [reply test/e2e/conformance/tests/httproute-consul-demo.go:25]: OK

File: test/e2e/conformance/tests/httproute-consul-demo.go (hunk 2 above)
Comment (2456868764) [reply test/e2e/conformance/tests/httproute-consul-demo.go]: OK

**Review by johnlanni — CHANGES_REQUESTED** [review]

**Review by johnlanni — APPROVED** [review]
LGTM

**Pull request merged**